edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -111,22 +99,23 @@ internal static string ValidateAndGetImage(string image, IEnvironmentWrapper env
         {
             image = Preconditions.CheckNonWhiteSpace(image, nameof(image)).Trim();
 
-            if (image.Substring(0, 1) == "$")
+            if ((image.Length > 0) && (image[0] == '$'))
             {
                 Match matchHost = ImageUpstreamRegex.Match(image);
                 if (matchHost.Success
-                    && (matchHost.Groups["host"]?.Length > 0)
                     && (matchHost.Groups["path"]?.Length > 0)
                     && (env != null))
                 {
-                    string enVarName = matchHost.Groups["host"].Value;
-
-                    string hostAddress = env.GetVariable(enVarName);
+                    string hostAddress = env.GetVariable(Gateway_hostname);
 
                     if (hostAddress != null)
                     {
                         image = hostAddress + matchHost.Groups["path"].Value;
                     }
+                    else
+                    {
+                        throw new ArgumentException($"Could not find environment variable: {Gateway_hostname}");
```

Comment:
we don't have any guide to throw what exception when a system parameter is missing.  But Argument exception should not be used as its definition is "The exception that is thrown when one of the arguments provided to a method is not valid.".  And this is not a value passing to this method.  I would suggest to use InvalidOperationException, please confirm with Varun.